source/server/http/admin.cc

```diff
@@ -575,13 +574,66 @@ std::string AdminImpl::runtimeAsJson(
   return strbuf.GetString();
 }
 
+Http::Code AdminImpl::handlerHystrixEventStream(const std::string&,
+                                                Http::HeaderMap& response_headers,
+                                                Buffer::Instance&, HandlerInfo& handler_info) {
+
+  response_headers.insertContentType().value().setReference(
+      Http::Headers::get().ContentTypeValues.TextEventStream);
+  response_headers.insertCacheControl().value().setReference(
+      Http::Headers::get().CacheControlValues.NoCache);
+  response_headers.insertConnection().value().setReference(
+      Http::Headers::get().ConnectionValues.Close);
+  response_headers.insertAccessControlAllowHeaders().value().setReference(
+      Http::Headers::get().AccessControlAllowHeadersValue.AccessControlAllowHeadersHystrix);
+  response_headers.insertAccessControlAllowOrigin().value().setReference(
+      Http::Headers::get().AccessControlAllowOriginValue.All);
+  response_headers.insertNoChunks().value().setReference("0");
+
+  HystrixHandlerInfo& hystrix_handler_info = dynamic_cast<HystrixHandlerInfo&>(handler_info);
+
+  // start streaming
+  hystrix_handler_info.data_timer_ = hystrix_handler_info.callbacks_->dispatcher().createTimer(
+      [this, &hystrix_handler_info]() -> void {
+        HystrixHandler::prepareAndSendHystrixStream(&hystrix_handler_info, server_);
+      });
+  hystrix_handler_info.data_timer_->enableTimer(
+      std::chrono::milliseconds(Stats::Hystrix::GetRollingWindowIntervalInMs()));
+
+  // start keep alive ping
+  hystrix_handler_info.ping_timer_ =
+      hystrix_handler_info.callbacks_->dispatcher().createTimer([&hystrix_handler_info]() -> void {
+        HystrixHandler::sendKeepAlivePing(&hystrix_handler_info);
+      });
+
+  hystrix_handler_info.ping_timer_->enableTimer(
+      std::chrono::milliseconds(Stats::Hystrix::GetPingIntervalInMs()));
+
+  ENVOY_LOG(debug, "start sending data to hystrix dashboard on port {}",
+            hystrix_handler_info.callbacks_->connection()->localAddress()->asString());
+  return Http::Code::OK;
+}
+
 void AdminFilter::onComplete() {
   std::string path = request_headers_->Path()->value().c_str();
   ENVOY_STREAM_LOG(debug, "request complete: path: {}", *callbacks_, path);
 
   Buffer::OwnedImpl response;
   Http::HeaderMapPtr header_map{new Http::HeaderMapImpl};
-  Http::Code code = parent_.runCallback(path, *header_map, response);
+  Http::Code code;
+  bool end_stream = true;
+
+  if (path.find("/hystrix_event_stream") == std::string::npos) {
+    HandlerInfoPtr temp_handler(new HandlerInfo);
+    handler_info_ = std::move(temp_handler);
+    code = parent_.runCallback(path, *header_map, response, *handler_info_);
+  } else {
+    HandlerInfoPtr temp_handler(new HystrixHandlerInfo(callbacks_));
```

Comment:
This is only half-baked, but maybe it makes sense to make all the handlers classes instead of just functions and pass them (or provide callbacks to get) certain state, like `callbacks_`, on each request.  Then, they can manage the creation of any per-request state that may need to be created internally.  That would eliminate the need to modify this function every time a handler wants to use HandlerInfo, and it would eliminate the `dynamic_cast`. 
 What do you think?

Reply:
I agree, we also thought about something like that. How about opening a separate issue for that?

Reply:
Yup, that seems reasonable to me.  Could you also toss a TODO in there referencing the issue?